app/services/hyrax/solr_service.rb

```diff
@@ -141,8 +115,23 @@ def search_by_id(id, opts = {})
 
     private
 
+      ##
+      # @private
+      # Return the valkyrie solr index.
+      #
+      # Since this module depends closely on RSolr internals and makes use
+      # of `#connection`, it will always need to connect to a Solr index. Other
+      # valkyrie indexers used here would, at minimum, need to provide a
+      # functioning `rsolr` connection.
       def valkyrie_index
-        Hyrax.index_adapter
+        Valkyrie::IndexingAdapter.find(:solr_index)
```

Comment:
I do agree that on the whole this service (and everything that calls it) has to be refactored before a elastic search or something indexer can be implemented, but I disagree with making this a pure solr lock here. What about a null indexer? it would be useful to start down the path of making this generic, not locking it even tighter to have to do more refactoring in the future.

Reply:
the issue i'm trying to avoid here is that `Hyrax.index_adapter` *can* vary (e.g. it may be a null adapter), but this one should always return an `rsolr` connection.

the pattern we've landed on for `Valkyrie::IndexingAdapter` is a ["write only adapter"](https://github.com/samvera/valkyrie/issues/598), and i guess the real issue here is that there's no corresponding pattern for the read side, other than "just use RSolr".

is the alternative to just leave this as `Hyrax.index_adapter`, accepting that this will raise errors when `use_valkyrie` is on and a non-RSolr powered indexer is configured?

Reply:
Ah, I see, I retract to comment then. I didn't fully understand the write only idea.